components/tidb_query/src/expr/builtin_string.rs

```diff
@@ -173,19 +173,32 @@ impl ScalarFunc {
         ctx: &mut EvalContext,
         row: &'a [Datum],
     ) -> Result<Option<Cow<'a, [u8]>>> {
+        use crate::expr::Expression;
+        fn collect_valid_strs<'a, 'b: 'a>(
+            exps: &'b [Expression],
+            ctx: &mut EvalContext,
+            row: &'a [Datum],
+        ) -> Result<Vec<Cow<'a, [u8]>>> {
+            let mut result = Vec::new();
+            for exp in exps {
+                let x = exp.eval_string(ctx, row)?;
+                if let Some(s) = x {
+                    result.push(s);
+                }
+            }
+            Ok(result)
+        }
```

Comment:
Although it's clear enough, you can use `iter` and `try_fold` to simplify your code.

Reply:
I'm not sure `try_fold` can work as expected. I plan to use `filter` to get rid of  all `NULL` from input. But the `Err` from `eval_string` needs to be exposed to the caller.

`try_fold` works on return value of closure, but we want side effect here(put elements into vec)

Reply:
OK, I will approve it no matter you change it or not.(But other reviewers may request you to change this)